### PR TITLE
Add an `inv setup` task to automatically check, configure and update the environment

### PR DESCRIPTION
**Pull request by FlorentClarret**
<!--
* New contributors are highly encouraged to read our
  [CONTRIBUTING](/CONTRIBUTING.md) documentation.
* Both Contributor and Reviewer Checklists are available at https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#pull-requests.
* The pull request:
  * Should only fix one issue or add one feature at a time.
  * Must update the test suite for the relevant functionality.
  * Should pass all status checks before being reviewed or merged.
* Commit titles should be prefixed with general area of pull request's change.
-->
### What does this PR do?

<!--
* A brief description of the change being made with this pull request.
* If the description here cannot be expressed in a succinct form, consider
  opening multiple pull requests instead of a single one.
-->

Add an `inv setup` task to automatically check, configure and update the environment. It currently: 

1. Checks the python version 
2. Checks the go version
3. Checks if the current branch is behind `origin/main` (just a warn if behind)
4. Updates the dependencies from the requirements file
5. Enables the pre-commit hooks

More actions could be added in the future (go-related or not).

### Motivation

<!--
* What inspired you to submit this pull request?
* Link any related GitHub issues or PRs here.
-->

It would be nice to have a script that:

1. Easily install the tools needed to work on the agent
2. Make sure you are using the latest version of everything

This script can do both. The idea is that if someone gets a weird error locally (e.g. module not found or whatever), instead of reaching out to our slack channel they start with `inv setup` and try again. This would for instance fix most of the issues related to new dependencies. Also they won't need to know exactly what's the command to run, since we will hide it in the script.

### Additional Notes

<!--
* Anything else we should know when reviewing?
* Include benchmarking information here whenever possible.
* Include info about alternatives that were considered and why the proposed
  version was chosen.
-->

Some examples (without color): 

All good: 

```
❯ inv setup
Checking Python version...                                                                                                                                                                                                                                                                                                                                        ─╯
Checking Go version...
Fetching git repository...
Checking main branch...
Updating dependencies...
Updating requirements.txt...
Updating tasks/requirements.txt...
Enabling pre-commit...

Results:

Check Python version     OK
Check Go version         OK
Check git repository     OK
Update dependencies from requirements.txt        OK
Update dependencies from tasks/requirements.txt  OK
Enable pre-commit        OK

Setup completed successfully.
```

One commit behind main:

```
❯ inv setup
Checking Python version...
Checking Go version...
Fetching git repository...
Checking main branch...
Updating dependencies...
Updating requirements.txt...
Updating tasks/requirements.txt...
Enabling pre-commit...

Results:

Check Python version     OK
Check Go version         OK
Check git repository     WARN
Your branch is 1 commit(s) behind main. Please update your branch.
Update dependencies from requirements.txt        OK
Update dependencies from tasks/requirements.txt  OK
Enable pre-commit        OK

Setup completed with warnings.
```

Wrong go version (I manually modified the `.go-version` file for this example):

```
❯ inv setup
Checking Python version...
Checking Go version...
Fetching git repository...
Checking main branch...
Updating dependencies...
Updating requirements.txt...
Updating tasks/requirements.txt...
Enabling pre-commit...

Results:

Check Python version     OK
Check Go version         FAIL
Go version is 1.21.9. Please install Go 1.21.8.
Check git repository     OK
Update dependencies from requirements.txt        OK
Update dependencies from tasks/requirements.txt  OK
Enable pre-commit        OK

Setup completed with errors.
```

### Possible Drawbacks / Trade-offs

<!--
* What are the possible side-effects or negative impacts of the code change?
-->

### Describe how to test/QA your changes

<!--
* Write here in detail or link to detailed instructions on how this change can
  be tested/QAd/validated, including any environment setup.
-->

### CODE REVIEW

**Review by amenasria — APPROVED** [review]
LGTM !

**👏 praise:** ‏This will be super useful !! Thanks for this !

**Review by pducolin — APPROVED** [review]
Yay!

**Review by pducolin — APPROVED** [review]

File: tasks/setup.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+    @staticmethod
+    def color(status):
+        if status == Status.OK:
+            return "green"
+
+        if status == Status.WARN:
+            return "orange"
+
+        return "red"
```
[the remark this answers is not in the document]
Comment (FlorentClarret) [reply tasks/setup.py:33]: Moved!

**Pull request merged**